src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -106,7 +122,12 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return _method.ToUpperInvariant().GetHashCode();
+            if (_hashcode == 0)
+            {
+                _hashcode = _method.ToUpperInvariant().GetHashCode();
```

Comment:
Using strings in a case statement doesn't seem fast.  What "equality" comparer will it use? Usually only numerics are put into a case statement.